circle.yml

```diff
@@ -4,29 +4,103 @@ machine:
+          mkdir -v -p "${CIRCLE_ARTIFACTS}/${DOCS_NGNIX}"
+          sudo mv -f ~/${CIRCLE_PROJECT_REPONAME}/_site $_
+          # Exit case statement if run in parallel else proceed to next case.
+          $CIRCLE_PARALLEL && exit 0
+          ;&
+        1)
+          printf "\n%s\n\n" "=== TEST ruby container with build and serve site ==="
+          docker build -t "${DOCKER_IMAGE_RUBY}:${DOCKER_IMAGE_TAG}" -f "${DOCKERFILE_RUBY}" .
+          docker run --name "${DOCS_RUBY}" -p "${DOCS_RUBY_BIND_ADDRESS}:4000" --volume=$(pwd):"${DOCS_RUBY_VOL}" -it -d "${DOCKER_IMAGE_RUBY}:${DOCKER_IMAGE_TAG}"
+          sleep 50
+          curl --silent "${DOCS_RUBY_BIND_ADDRESS}" | grep -q "blog" && print " ---> Served jekyll site is OK"
+          docker stop "${DOCS_RUBY}"
+          printf "\n%s\n\n" "=== SUCCESS ==="
+          printf "\n%s\n\n" "=== ARCHIVE site artifacts of ruby container ==="
+          mkdir -v -p "${CIRCLE_ARTIFACTS}/${DOCS_RUBY}"
+          sudo mv -f ~/${CIRCLE_PROJECT_REPONAME}/_site $_
+          ;;
+        esac
+      : parallel: true
+
 
 deployment:
-  dockerimages:
+  push_tag_dev_master:
     branch:
       - master
       - develop
     commands:
-      - docker run --name elasticsearch -p 9200:9200 -d choffmeister/elasticsearch-prefilled:latest
-      - sleep 10
-      - bundle exec rake index
-      - sleep 10
-      - docker stop elasticsearch
-
-      - docker login -u "$DOCKER_LOGIN_USERNAME" -p "$DOCKER_LOGIN_PASSWORD" -e "$DOCKER_LOGIN_EMAIL" docker.epages.com
-      - docker commit elasticsearch "$DOCKER_IMAGE_ELASTICSEARCH:${CIRCLE_BRANCH//\//-}"
-      - docker push "$DOCKER_IMAGE_ELASTICSEARCH:${CIRCLE_BRANCH//\//-}"
-
-      - bundle exec rake build
-      - docker build -t "$DOCKER_IMAGE_NGINX:${CIRCLE_BRANCH//\//-}" .
-      - docker push "$DOCKER_IMAGE_NGINX:${CIRCLE_BRANCH//\//-}"
+      - docker login -u "${DOCKER_LOGIN_USERNAME}" -p "${DOCKER_LOGIN_PASSWORD}" -e "${DOCKER_LOGIN_EMAIL}" "${DOCKER_REGISTRY}"
+      - docker push "${DOCKER_IMAGE_ELASTICSEARCH}:${DOCKER_IMAGE_TAG}"
+      - docker push "${DOCKER_IMAGE_NGINX}:${DOCKER_IMAGE_TAG}"
+      - docker push "${DOCKER_IMAGE_RUBY}:${DOCKER_IMAGE_TAG}"
+  push_tag_stable_latest:
+    branch: master
+    commands:
+      - docker login -u "${DOCKER_LOGIN_USERNAME}" -p "${DOCKER_LOGIN_PASSWORD}" -e "${DOCKER_LOGIN_EMAIL}" "${DOCKER_REGISTRY}"
+      - docker tag "${DOCKER_IMAGE_NGINX}:${DOCKER_IMAGE_TAG}" "${DOCKER_IMAGE_NGINX}:stable"
+      - docker tag "${DOCKER_IMAGE_NGINX}:${DOCKER_IMAGE_TAG}" "${DOCKER_IMAGE_NGINX}:latest"
+      - docker tag "${DOCKER_IMAGE_RUBY}:${DOCKER_IMAGE_TAG}" "${DOCKER_IMAGE_RUBY}:stable"
+      - docker tag "${DOCKER_IMAGE_RUBY}:${DOCKER_IMAGE_TAG}" "${DOCKER_IMAGE_RUBY}:latest"
+      - docker push "${DOCKER_IMAGE_NGINX}:stable"
+      - docker push "${DOCKER_IMAGE_NGINX}:latest"
+      - docker push "${DOCKER_IMAGE_RUBY}:stable"
+      - docker push "${DOCKER_IMAGE_RUBY}:latest"
```

Comment:
I am quite sure, that at most one deployment configuration is executed. Means that when pushing master, then only `push_tag_dev_master` is executed, even though `push_tag_stable_latest` would match, too. But first one wins.